.github/isbin.sh

```diff
@@ -6,27 +6,32 @@ size_threshold=65536
 
 SAVEIFS=$IFS
 IFS=$(echo -en "\n\b")
+OS=$(uname -a | cut -d ' ' -f 1)
 
 for f in $@; do
     echo $f
-    size=$(stat --format %s $f)
+    case "$OS" in
+        Linux)  size=$(stat --format "%s" $f);;
+        Darwin) size=$(stat -f "%z" $f);;
+        *)      echo "Error: unsupported operating system $OS" >&2; exit 1;;
+    esac
     if [ $size -gt $size_threshold ]; then
-     echo "file exceeds maximum allowable size"
-     echo "raw data and ipynb files should go in scico-data"
-        exit 1
+        echo "file exceeds maximum allowable size"
+        echo "raw data and ipynb files should go in scico-data"
+        exit 2
     fi
     charset=$(file -b --mime $f | sed -e 's/.*charset=//')
     if [ ! -L "$f" ] && [ "$charset" = "binary" ]; then
-     echo "binary files cannot be commited to the repository"
-     echo "raw data and ipynb files should go in scico-data"
-        exit 2
+        echo "binary files cannot be commited to the repository"
+        echo "raw data and ipynb files should go in scico-data"
```

Comment:
Is this always the case?

Reply:
Yes, with a very small number of exceptions (such as `scico.ico`).